src/main/java/thrift/model/Model.java

```diff
@@ -71,11 +72,21 @@
      */
     void addExpense(Expense expense);
 
+    /**
+     * Adds the given expense to a specified index.
+     */
+    void addExpense(Expense expense, Index index);
```

Comment:
Nice way of handling `undo` a `delete` command by creating a new constructor that accepts index-based adding!